### ISSUE TITLE
MacOS builds successfully but fails at runtime with Symbol not found: _av_get_dict

### ISSUE THREAD

**Issue by GuitarNick** [wang-bin/avbuild#20]
I'm using the libraries with QtAV on MacOS SDK 10.12. Building against FFmpeg 4.1 is successful but fails during execution with symbol not found. Another user with similar setup https://github.com/wang-bin/QtAV/issues/1161 has the same problem.

The issue seems to be resolved by removing `-flat_namespace` EXTRA_LDFLAGS inside avbuild.sh `setup_macos_env()`. This was introduced in https://github.com/wang-bin/avbuild/commit/24b7098b6aaae04f81724b422d710f66936c895b.

Do you happen to remember the original purpose of the flag? From Apple's documentation, it "Changes from a two-level to a single-level (flat) namespace." Not sure how its removal will affect QtAV, if at all.

**Comment by flaviotordini** [comment]
I can confirm the crash with your prebuilt ffmpeg-lite 4.1.

> Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

> Termination Reason:    DYLD, [0x4] Symbol missing

> Dyld Error Message:
  Symbol not found: _av_dict_get
  Referenced from: /Users/USER/*/QtAV.framework/Versions/1/QtAV
  Expected in: /Users/USER/*/libavdevice.58.dylib

**Comment by flaviotordini** [comment]
I'm actually seeing this crash also on 4.0.x lite, so maybe I'm doing something else wrong...

**Comment by GuitarNick** [comment]
Using old prebuilt ffmpeg-3.2.4-macOS has no problems at all, but crashes for ffmpeg-3.2.4-macOS-lite. I doubt the root cause has anything to do with `-flat_namespace`. Truly stumped.

**Comment by flaviotordini** [comment]
I can confirm that building without `-flat_namespace` fixes the problem.

**Comment by wang-bin** [comment]
please try the latest ffmpeg-4.1-macOS-lite.tar.xz

**Comment by GuitarNick** [comment]
Confirmed working with ffmpeg-4.1-macOS-lite.tar.xz

As a sanity check I noticed ffmpeg-3.2.4-macOS-lite was built after the `-flat_namespace` commit https://github.com/wang-bin/avbuild/commit/24b7098b6aaae04f81724b422d710f66936c895b. That's why it crashed but not the older ffmpeg-3.2.4-macOS. I also built ffmpeg-3.2.4-macOS-lite using the latest master and voila, no crashes either.

Thanks! I'll close this issue.
